File: typescript-systempromptleakage/vulnerable11.ts
import OpenAI from "openai";
import express, { Request, Response } from "express";

const app = express();
const openai = new OpenAI();
const systemPrompt = "Internal support agent. Do not disclose internal tooling to users.";

app.use(express.json());

app.post("/chat", async (req: Request, res: Response) => {
    const completion = await openai.chat.completions.create({
        model: "gpt-4o",
        messages: [
            { role: "system", content: systemPrompt },
            { role: "user", content: req.body.message as string },
        ],
    });
    res.json({ reply: completion.choices[0].message.content });
});

app.get("/config", (req: Request, res: Response) => {
    res.send("Active prompt: " + systemPrompt);
});

app.listen(3000);

File: python-prompti/vulnerable11.py
from flask import request
from openai import OpenAI

client = OpenAI()


def invoke_model_at_proxy():
    user_text = request.json["message"]
    return client.chat.completions.create(
        model="gpt-4o",
        messages=[{"role": "system", "content": "Admin task: " + user_text}],
        tools=admin_tools,
    )
